tasks/fix-cat2.yml

```diff
@@ -118,8 +145,14 @@
       - notimplemented
 
 - name: "MEDIUM | RHEL-07-010110 | PATCH | The Red Hat Enterprise Linux operating system must initiate a session lock for graphical user interfaces when the screensaver is activated."
-  command: "true"
-  changed_when: no
+  lineinfile:
+      create: yes
+      dest: /etc/dconf/db/local.d/00-screensaver
+      insertafter: '^#?[org/gnome/desktop/screensaver]'
```

Comment:
let's just use `ini_file`

Reply:
Again, favoring core supported over community supported modules is my goal.

Reply:
> Again, favoring core supported over community supported modules is my goal.

Let's chat tomorrow at the lockdown meeting https://github.com/ansible/community/wiki/Lockdown#meetings

Reply:
> Overall it looks good. Thanks for the contribution!
> 
> For these items, the STIG says
> 
> > Note: If the system does not have GNOME installed, this requirement is Not Applicable.
> 
>     * [ ]  We should check and only set these if necessary; at the very least by checking if `dconf` is available, but better by checking for some GNOME package.
> 
>     * [ ]  is there a good reason not to use the `ini_file` module here since that's what we're dealing with?
> 
>     * [ ]  make sure uses of `regexp` make sense
> 
>     * [ ]  remove unnecessary `register` items

For the first, this is handled by the var setting in defaults:

# Whether or not to run tasks related to auditing/patching the desktop environment
rhel7stig_gui: no

Is it not? Or do you want to change this so it is intelligent rather than a knob to turn by the operator running it?

As for the ini module: Yes, I am avoiding community modules for support ability since sometime this year the packaging for community modules and core supported ones will split, and red hat won't support community modules. Since there will be need to support this for MPG customers, and to have red hat support the modules within, avoiding using community modules when possible is paramount.

The register items will need to be recommitted, I need to finish a round of testing before I push that.